http/logical.go

```diff
@@ -200,8 +201,14 @@ func respondRaw(w http.ResponseWriter, r *http.Request, resp *logical.Response)
 		retErr(w, "no status code given")
 		return
 	}
-	status, ok := statusRaw.(int)
-	if !ok {
+
```

Comment:
For safety, you should add a `json.Number` check in here too, in case something has already serialized the data for some reason. It would be handled transparently by the JSON encoder for the response but would trip this here.